src/plugins/mvcc_delete_plugin.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <algorithm>
+#include <mutex>
+#include <numeric>
+#include <queue>
+#include <thread>
+
+#include "gtest/gtest_prod.h"
+#include "storage/chunk.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/abstract_plugin.hpp"
+#include "utils/pausable_loop_thread.hpp"
+#include "utils/singleton.hpp"
+
+namespace opossum {
+
+/*
+ * One disadvantage of insert only databases like Hyrise is the accumulation of invalidated
+ * rows, which have to be removed from the final result for every transaction.
+ * This plugin deletes chunks with large numbers of invalidated rows. Thus, it keeps the
```

Comment:
... and reinserts them at the end of the table. These rows are either visible at their original position (for old transactions) or their new position (for new transactions)